Encounter an compiler error after translate a TS lib to assemblyscript

Here is the error info:
/Users/dean/assemblyscript/std/portable/index.js:198
     throw new AssertionError(message);
           ^
 
 AssertionError: assertion failed
     at null.Y.assert (/Users/dean/assemblyscript/std/portable/index.js:198:11)
     at sr.visitElement (/Users/dean/assemblyscript/src/bindings/util.ts:116:16)
     at sr.visitFile (/Users/dean/assemblyscript/src/bindings/util.ts:58:14)
     at sr.visitFile (/Users/dean/assemblyscript/src/bindings/util.ts:65:14)
     at sr.walk (/Users/dean/assemblyscript/src/bindings/util.ts:46:64)
     at sr.build (/Users/dean/assemblyscript/src/bindings/tsd.ts:201:10)
     at Function.build (/Users/dean/assemblyscript/src/bindings/tsd.ts:41:41)
     at Module.XS (/Users/dean/assemblyscript/src/index-wasm.ts:363:21)
     at Module.Mn (/Users/dean/assemblyscript/cli/index.js:974:35)
     at async file:///Users/dean/assemblyscript/bin/asc.js:33:22

I'am trying to Translate https://github.com/EsotericSoftware/spine-runtimes/tree/4.1/spine-ts/spine-core to assemblyscript. Some moduler like Texutre and AssetManager has been removed and hunders of error has been fixed, but still get lots of warnings and all of them are like:"WARNING AS235: Only variables, functions and enums become WebAssembly module exports.".

--- a/src/bindings/util.ts
+++ b/src/bindings/util.ts
@@ -19,7 +19,8 @@ import {
   Global,
   Program,
   Property,
-  PropertyPrototype
+  PropertyPrototype,
+  InterfacePrototype
 } from "../program";
 
 /** Walker base class. */
@@ -94,6 +95,10 @@ export abstract class ExportsWalker {
         this.visitClassInstances(name, <ClassPrototype>element);
         break;
       }
+      case ElementKind.InterfacePrototype: {
+        this.visitInterfaceInstances(name, <InterfacePrototype>element);
+        break;
+      }
       case ElementKind.PropertyPrototype: {
         let propertyInstance = (<PropertyPrototype>element).instance;
         if (!propertyInstance) break;
@@ -112,8 +117,13 @@ export abstract class ExportsWalker {
         if (hasCompiledMember(element)) this.visitNamespace(name, element);
         break;
       }
-      case ElementKind.TypeDefinition: break;
-      default: assert(false);
+      case ElementKind.TypeDefinition:
+      case ElementKind.IndexSignature: break;
+      default: {
+        // Not (directly) reachable exports:
+        // File, Local, Function, Class, Interface
+        assert(false);
+      }
     }
   }
 
@@ -134,7 +144,20 @@ export abstract class ExportsWalker {
       // TODO: for (let instance of instances.values()) {
       for (let _values = Map_values(instances), i = 0, k = _values.length; i < k; ++i) {
         let instance = unchecked(_values[i]);
+        assert(instance.kind == ElementKind.Class);
         if (instance.is(CommonFlags.Compiled)) this.visitClass(name, instance);
+      }
+    }
+  }
+
+  private visitInterfaceInstances(name: string, element: InterfacePrototype): void {
+    let instances = element.instances;
+    if (instances) {
+      // TODO: for (let instance of instances.values()) {
+      for (let _values = Map_values(instances), i = 0, k = _values.length; i < k; ++i) {
+        let instance = <Interface>unchecked(_values[i]);
+        assert(instance.kind == ElementKind.Interface);
+        if (instance.is(CommonFlags.Compiled)) this.visitInterface(name, instance);
       }
     }
   }

--- a/tests/compiler/bindings/esm.ts
+++ b/tests/compiler/bindings/esm.ts
@@ -169,3 +169,10 @@ immutableGlobalNested;
 declare function Date_getTimezoneOffset(): i32;
 
 Date_getTimezoneOffset();
+
+// Not yet instrumented element kinds:
+
+export class ExportedClass {}
+export interface ExportedInterface {}
+export type ExportedType = ExportedClass;
+export namespace ExportedNamespace {}

